search-service/src/main/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationService.kt

```diff
@@ -171,32 +170,28 @@ class EnabledAuthorizationService(
         }
     }
 
-    override suspend fun computeAccessRightFilter(sub: Option<Sub>): () -> String? {
-        if (subjectReferentialService.hasStellioAdminRole(sub).getOrElse { false })
-            return { null }
-        else {
-            return subjectReferentialService.getSubjectAndGroupsUUID(sub)
-                .map {
-                    if (it.isEmpty()) {
-                        { "(specific_access_policy = 'AUTH_READ' OR specific_access_policy = 'AUTH_WRITE')" }
-                    } else {
-                        {
-                            """
-                            ( 
-                                (specific_access_policy = 'AUTH_READ' OR specific_access_policy = 'AUTH_WRITE')
-                                OR
-                                (entity_payload.entity_id IN (
-                                    SELECT entity_id
-                                    FROM entity_access_rights
-                                    WHERE subject_id IN (${it.toListOfString()})
-                                ))
-                            )
-                            """.trimIndent()
-                        }
-                    }
-                }.getOrElse { { "1 = 0" } }
-        }
-    }
+    override suspend fun computeAccessRightFilter(sub: Option<Sub>): () -> String? =
+        subjectReferentialService.getSubjectAndGroupsUUID(sub).map { uuids ->
+            if (uuids.isEmpty()) {
```

Comment:
don't remember why there is this check but it is quite weird...

Reply:
because after we have checked a subject id in the list. But if it is empty there is an error

Reply:
Yes, I know this. My point was: how is it possible that the `getSubjectAndGroupsUUID()` function returns an empty list?

Reply:
It even seems that a Left is returned if no result found in `getSubjectAndGroupsUUID()`

Reply:
(ah did not see the thumb here - I get no notification for this :) - that answers my first question in the PR)